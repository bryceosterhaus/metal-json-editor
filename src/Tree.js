import Component, {Config} from 'metal-jsx';

import Node from './Node';

class Tree extends Component {
	render() {
		const {
			arrowRenderer,
			data,
			expandedNodes,
			locator,
			onChange,
			onToggleExpand
		} = this.props;

		return (
			<ul class="tree-container" style="list-style: none; margin: 0;">
				{
					Object.keys(data).map(
						key => (
							<Node
								arrowRenderer={arrowRenderer}
								expandedNodes={expandedNodes}
								locator={[...locator, key]}
								name={key}
								onChange={onChange}
								onToggleExpand={onToggleExpand}
								value={data[key]}
							/>
						)
					)
				}
			</ul>
		);
	}
}

Tree.PROPS = {
	arrowRenderer: Config.func().value(() => {}),
	data: Config.oneOfType(
		[
			Config.array(),
			Config.object()
		]
	).value({}),
	expandedNodes: Config.instanceOf(WeakSet).required(),
	onChange: Config.func(),
	onToggleExpand: Config.func(),
	locator: Config.array().value([])
};

export default Tree;
