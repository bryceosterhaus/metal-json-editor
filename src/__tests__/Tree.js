import Tree from '../Tree';

describe('Tree', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('renders', () => {
		component = new Tree({
			data: {
				one: 2,
				three: 'four'
			},
			expandedNodes: new WeakSet()
		});

		expect(component).toMatchSnapshot();
	});
});
