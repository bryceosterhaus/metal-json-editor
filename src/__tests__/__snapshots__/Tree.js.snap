// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree renders 1`] = `
<ul class="tree-container" style="list-style: none; margin: 0px;">
  <li key="_metal_jsx_1">
    <div key="_metal_jsx_1">
      <span style="" key="_metal_jsx_1">one: </span>
      <span style="color: rgb(28, 0, 207);" title="Click to Edit" key="_metal_jsx_2">
        <span style="white-space: pre;" key="_metal_jsx_1">2</span>
      </span>
    </div>
  </li>
  <li key="_metal_jsx_2">
    <div key="_metal_jsx_1">
      <span style="" key="_metal_jsx_1">three: </span>
      <span style="color: rgb(196, 26, 22);" title="Click to Edit" key="_metal_jsx_2">
        <span style="white-space: pre;" key="_metal_jsx_1">four</span>
      </span>
    </div>
  </li>
</ul>
`;
